### ISSUE TITLE
[Accessibility][HC]: Incorrect DataGridView ComboBox and TextBox cells HC1/2 color

### ISSUE THREAD

**Issue by vladimir-krestov** [dotnet/winforms#1953]
* .NET Core Version: 3.1.0-preview1.19458.7
* Have you experienced this same bug with .NET Framework?: Yes

**Original bug:**
- [Bug 820338:](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/820338) Accessibility: [MAS42A] [HC 1/2]: The text color is incorrect in DGV on HC 1/2 theme

**Problem description:**
- A user gets incorrect text color of DataGridView ComboBox and TextBox cells which should be like as in Windows settings:
![image](https://user-images.githubusercontent.com/49272759/65384026-ef4f7080-dd25-11e9-8d8f-b22630b4bd0e.png)

**Actual behavior:** 
- DataGridViewComboBox and DataGridViewTextBox cells text color is not like as in Windows HC theme
![image](https://user-images.githubusercontent.com/49272759/65384266-f035d180-dd28-11e9-9b1c-9ef74dfea6d1.png)

**Expected behavior:**
- DataGridViewComboBox and DataGridViewTextBox cells text color should be like as in Windows HC theme
![image](https://user-images.githubusercontent.com/49272759/65384372-2de72a00-dd2a-11e9-84b6-b1d4a3d0a92d.png)

**Minimal repro:**
1. Create .Net Core WinForms app
2. Add a DataGridView to the app 
3. Add a DataGridViewComboBoxColumn and a DataGridViewTextBoxColumn to the DataGridView
4. Turn on HighContrast theme in Windows settings
5. Check text color of DataGridViewComboBox and DataGridViewTextBox cells

**Comment by merriemcgaw** [comment]
I'm removing the a11yMAS label because this wasn't part of a test pass, and the contrast ratio should be just fine as is. If we get an official bug on it of couse we will look into it.

**Comment by NikitaSemenovAkvelon** [comment]
In Win11 HC themes are different. And data inside DGV looks different too.
Default style:
![image](https://user-images.githubusercontent.com/109065597/182787920-4c03f024-0099-4ee7-bdb0-4a9dc93d275f.png)
How it looks in all HC themes:

1. Aquatic 
![image](https://user-images.githubusercontent.com/109065597/182788250-b2764b2f-40e6-4793-aeb7-78745baaf66f.png)
2. Desert 
![image](https://user-images.githubusercontent.com/109065597/182788458-b3728b74-df49-4ba2-bd38-7d19a9a01400.png)
3. Dusk 
![image](https://user-images.githubusercontent.com/109065597/182788560-8b4021a5-a8d5-41ea-88ed-5410a00b9e58.png)
4. Night sky 
![image](https://user-images.githubusercontent.com/109065597/182788666-ec8316db-38bd-4220-aab7-d0d74a03fc4c.png)

So, using HC themes all values in the rows have the same color.

**Comment by NikitaSemenovAkvelon** [comment]
Using manifest with settings for Win10/Win11 some changes from #1966 (https://github.com/dotnet/winforms/pull/1966/files#diff-9d504543bd501e09991d02bff89a815d6245eeeb12fe3dc82da960caebc9c5a1R2371, https://github.com/dotnet/winforms/pull/1966/files#diff-cd5392ce149aed5451edcee98c4573469d3901139bf20d96e999838223683a06R914) fixes the issue. Now, it looks correctly (DataGridViewTextBoxColumn like TextBox, DataGridViewComboBoxColumn like ComboBox and DataGridViewButtonColumn like Button):
![image](https://user-images.githubusercontent.com/109065597/183610226-1b1ec1c4-f886-424b-b6fc-b2aeebc606f4.png)

**Comment by Philip-Wang01** [comment]
Verified this issue in the latest .Net 8.0 SDK build: 8.0.100-alpha.1.22478.15, it was fixed.please check below screenshot:
Win10:
![image](https://user-images.githubusercontent.com/86937911/192976074-3bb2abf1-40f2-41b1-9405-0ef4287a3bcb.png)
Win11:
![image](https://user-images.githubusercontent.com/86937911/192977072-c79c44b7-eb5e-4a7a-bcdc-87078d289f16.png)
